service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ChildControllerMidEventTest.java

```diff
@@ -0,0 +1,219 @@
+@WebMvcTest(ChildController.class)
+@OverrideAutoConfiguration(enabled = true)
+class ChildControllerMidEventTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String ERROR_MESSAGE = "Date of birth cannot be in the future";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldReturnErrorWhenFirstChildDateOfBirthIsInFuture() throws Exception {
+        ZonedDateTime today = ZonedDateTime.now();
+        ZonedDateTime tomorrow = today.plusDays(1);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(
+            new OldChildren(createChild(tomorrow), createChild(today))
+        );
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnErrorWhenAdditionalChildDateOfBirthIsInFuture() throws Exception {
+        ZonedDateTime today = ZonedDateTime.now();
+        ZonedDateTime tomorrow = today.plusDays(1);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(
+            new OldChildren(createChild(today), createChild(tomorrow))
+        );
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnNoErrorsWhenAllDatesOfBirthAreTodayOrInPast() throws Exception {
+        ZonedDateTime today = ZonedDateTime.now();
+        ZonedDateTime yesterday = today.minusDays(1);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(
+            new OldChildren(createChild(today), createChild(yesterday))
+        );
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnDateOfBirthErrorsForNewChildWhenFutureDateOfBirth() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of(
+                    "children1", ImmutableList.of(
+                        ImmutableMap.of(
+                            "id", "",
+                            "value", Child.builder()
+                                .party(ChildParty.builder()
+                                    .dateOfBirth(Date.from(ZonedDateTime.now().plusDays(1).toInstant()))
+                                    .build())
+                                .build()
+                        )
+                    )
+                ))
+                .build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnDateOfBirthErrorsForNewChildrenWhenThereIsMultipleChildren() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
```

Comment:
Couldn't we have some helper methods for building the the "core" of the case and just pass children as the elements? A lot repeated code, builds etc.